### PR TITLE
#161966976 Users to view and edit their profiles

### PR DESCRIPTION
**Pull request by fabzer0**
#### What does this PR do?
    - This Pull request allows users to view and edit their own profiles.

#### Description of Task to be completed?
    - Get users profiles from the API and display to the browser.
    - Make a put request to the browser to edit the profile

#### How should this be manually tested?
```git clone https://github.com/andela/ah-jumanji-frontend.git```
```cd into the created folder <ah-jumanji-frontend>```
```git checkout ft-user-profile-161966976```
run ```npm i && npm audit fix && npm start```
Register to our site and then log in
Navigate to the profile page and click ```Edit Profile```
Upload images and edit the other credentials at your own will.

#### Any background context you want to provide?
Use of JSX/React/Redux in the frontend

#### What are the relevant pivotal tracker stories?
[161966976](https://www.pivotaltracker.com/story/show/161966976)

#### Screenshots (if appropriate)
screenshot 2019-01-29 at 09 44 15

#### Checklist:

- [x] My code follows the style guidelines of this project
- [ ] At least 2 people have reviewed my PR
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] My PR has one commit.

### CODE REVIEW

File: src/apps/Dashboard/actions/dashboardActions.js [hunk 1]
```diff
@@ -43,9 +43,9 @@ export function fetchArticle() {
             for (const i in r) {
               let slug = r[i].slug;
               // add likes, comments and rating of each article in the payload
-              axios.all([fetchArticleLikes(slug), fetchArticleComments(slug), fetchArticleRating(slug)]).then(
+              axios.all([fetchArticleLikes(slug), fetchArticleComments(slug), fetchArticleRating(slug), fetchProfiles()]).then(
                 axios.spread(
-                  (likes, comments, ratings) =>{
+                  (likes, comments, ratings, profiles) =>{
```
Comment (brayovsky) [inline src/apps/Dashboard/actions/dashboardActions.js]: Why add an argument that you are not using?

File: src/apps/UserProfile/components/countries.js [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+  "VG": "VIRGIN ISLANDS, BRITISH",
+  "VI": "VIRGIN ISLANDS, U.S.",
+  "WF": "WALLIS AND FUTUNA",
+  "EH": "WESTERN SAHARA",
+  "YE": "YEMEN",
+  "ZM": "ZAMBIA",
+  "ZW": "ZIMBABWE",
+};
+
+export default countries;
```
Comment (brayovsky) [inline src/apps/UserProfile/components/countries.js]: Add a new line

**Review by brayovsky — CHANGES_REQUESTED** [review]
Please also ignore .DS_Store file

**Comment by GransonO** [comment]
@fabischolasi the code climate issues are quite a lot. Consider refactoring to reduce the issues.

File: src/apps/UserProfile/components/countries.js (hunk 2 above)
Comment (fabzer0) [reply src/apps/UserProfile/components/countries.js]: resolved

File: src/apps/Dashboard/actions/dashboardActions.js (hunk 1 above)
Comment (fabzer0) [reply src/apps/Dashboard/actions/dashboardActions.js]: That was supposed to be done away with. Resolved

**Pull request merged**